packages/jest-snapshot/src/State.js

```diff
@@ -158,7 +158,7 @@ class SnapshotState {
       if (!pass) {
         this.unmatched++;
         return {
-          actual: receivedSerialized,
+          actual: receivedSerialized.replace(/\\"/g, '"'),
           count,
           expected,
```

Comment:
the tricky thing is this one. we only have the serialized version. The way we unserialize stuff right now is to eval. So we'd need to eval the expected part.